### DISCUSSION TITLE
Returning unit

### DISCUSSION THREAD

**Discussion by MichaelXavier** [chrisdone/hell#50] [Q&A] [answered]
I'm probably missing something obvious but I don't seem to be able to return `()` from a function. Sometimes I'll have a conditional that prints a message in one case and runs an effect that returns a value that I don't need in the other. There's no `Monad.void` and `Monad.return ()` gives me `Invalid variable: ()`. My workaround is to do `Text.putStr ""` which is kind of weird.

**Comment by chrisdone** [comment, marked as answer]
Agree, it’s just an oversight. On paper it should just be a case of adding () to the list of supported types and then desugaring it appropriately.
